test/evp_test.c

```diff
@@ -2172,11 +2063,55 @@ static int keypair_test_run(struct evp_test *t)
     return rv;
 }
 
-static const struct evp_test_method keypair_test_method = {
+static const evp_test_method keypair_test_method = {
     "PrivPubKeyPair",
     keypair_test_init,
     keypair_test_cleanup,
     void_test_parse,
     keypair_test_run
 };
 
+static const char *testfile;
+
+static int run_file_tests()
+{
+    BIO *in;
+    char buf[10240];
+    evp_test t;
+
+    if (!TEST_ptr(in = BIO_new_file(testfile, "rb")))
+        return 0;
+    memset(&t, 0, sizeof(t));
+    t.start_line = -1;
+    t.in = in;
+    t.err = NULL;
+    while (BIO_gets(in, buf, sizeof(buf))) {
+        t.line++;
+        if (!process_test(&t, buf, 0))
+            return 0;
+    }
+    /* Run any final test we have */
+    if (!evpsetup_test(&t, NULL))
+        return 0;
+
+    TEST_info("Completed %d tests with %d errors and %d skipped",
+              t.ntests, t.errors, t.nskip);
+    free_key_list(t.public);
+    free_key_list(t.private);
+    BIO_free(t.key);
+    BIO_free(in);
+    return t.errors == 0;
+}
+
+int test_main(int argc, char *argv[])
+{
+    if (argc != 2) {
+        TEST_error("Wrong argument count");
+        return 0;
+    }
+    testfile = argv[1];
+
+    ADD_TEST(run_file_tests);
```

Comment:
Side note: we might have to rethink the parsing of this to load everything into a test dict and ADD_ALL_TESTS it with the number of tests found.  I know that I argued exactly *against* this in your parser PR, but I just realised now how reading it all first could be benefitial for information on what's going on.

I'll repeat, this is a side note, not meant to affect this PR, but to inspire future changes.